Gems/EditorPythonBindings/Code/Source/PythonSystemComponent.cpp

```diff
@@ -258,6 +258,85 @@ namespace EditorPythonBindings
         void Finalize() override {}
     };
 
+    // Manages the acquisition and release of the Python GIL (Global Interpreter Lock).
+    // Used by PythonSystemComponent to lock the GIL when executing python.
+    class PythonSystemComponent::PythonGILScopedLock final
+    {
+    public:
+        PythonGILScopedLock(AZStd::recursive_mutex& lock, int& lockRecursiveCounter, bool tryLock = false);
+        ~PythonGILScopedLock();
+
+        bool IsLocked() const;
+
+    protected:
+        void Lock(bool tryLock);
+        void Unlock();
+
+        AZStd::recursive_mutex& m_lock;
+        int& m_lockRecursiveCounter;
+        bool m_locked = false;
```

Comment:
this may need to be std::atomic to make sure that there is exclusive access to the cache line, otherwise it could happen that two threads will see different value of m_locked